### DISCUSSION TITLE
Welcome to MyFinances Discussions!

### DISCUSSION THREAD

**Discussion by TreyWW** [TreyWW/MyFinances#28] [Ideas]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

More coming soon

**Comment by AndycLiang** [comment]
Hey there! I'm Andy, a first-year student in college. I was looking around for projects to get experience in programming and development, and to contribute to something I may find personally useful, and MyFinance seems interesting! I have some very limited experience in Python, and I'm planning on reading a book on C, and also using the Linux command line. I'm wondering if I could contribute to the project, and I'll be learning a lot.

**Comment by TreyWW** [comment]
Hey @AndycLiang,
Welcome along - I'm glad you found this project.
That's okay, our project hopefully allows for any skill level to help contribute, so i'll list the things we use in a beginner friendly way.
The main tools we use for this project are as follows:
- Python, this is our main [backend](https://www.techslang.com/definition/what-is-backend-development/) language
    - With python we use [Django](https://www.djangoproject.com/), which is our backend framework. This handles things like loading the pages and handling requests.
    - We use [Django UnitTest](https://docs.djangoproject.com/en/5.0/internals/contributing/writing-code/unit-tests/) (a fork/django version of [Py Unittest](https://docs.python.org/3/library/unittest.html).
 - HTML, this is our main "Styling/structure language" which controls the structure of the web page
     - [Tailwind CSS](https://tailwindcss.com/) is our main CSS Library (css is the actual styles and design of the HTML)
     - [DaisyUI](https://daisyui.com/) is our main Tailwind Component Library, this is good as it comes with many nice styles
 
Now as for things you could contribute to, it's up to you! Any of these things is open for contribution, however big or small. We don't mind if you help come up with ideas, find bugs, or add features. It's up to you, I can help you find something if you're not sure, but i'd suggest having a look at our [issues](https://github.com/TreyWW/issues) and seeing how they look.

Thanks again!

**Comment by HessTaha** [comment]
Hey there ! 

I've been working with python for a couple of years and I'd like to get involved with opensource 😄. 

I have some ideas for improving the packaging and styling of the project. (I can understand if its not a priority for now 😆 )

For the python part of the project : 
- Poetry for managing dependencies 
- Add pre-commit hooks and rules for code formatting 
- Enhancing  CI workflow ( for ex: add some isort to the current black )

For the documentation: 
- Mkdocs + github pages can be a good option for easy markdown documentation , it can also be automated with github actions

Global scope: 
- Having a Makefile or start.sh or a script to setup/start mysql, install the CSS framework , basically a "magic" kinda scripts that gets you set up super fast? I don't know how hard it can be to create this script but I believe it could be helpful😄 


What do you think? 
Also I'm new to django but I'm open to work on features/issues, if you have any issue/feature to prioritize, let me know and we can discuss it 😄

**Reply by TreyWW** [reply]
That's amazing, thank you for the ideas @HessTaha!
Poetry sounds good, i'll take a look into that, could really help with dependencies. Pre-commit hooks would too be great, but I tried them on an older project and they kind of just kept breaking things. Maybe was a bad setup or something, but we could try them for formatting!

As for documentation I've recently implemented Docsify which I'm quite liking as it keeps a nice markdown format and a lovely design. You mentioned they could be automated which I'm wondering how that would work, that could be quite useful!

Some scripts for automating setup, databases etc would too be good. Might be hard to work for multiple OS's and environment setups but I'll look into makefiles.

Overall they're some incredible ideas, and would benefit the project vastly if we added these. I'm happy to talk about them further if you have any more details on them, we can make Issues for each and then others can also contribute to each mini feature and then we'll get lots of great scripts made. 

Thanks again, really appreciate your contribution and discussion

**Reply by HessTaha** [reply]
Hey!
I'll look into pre commits and doc automation if that sounds good for you 😄 

Will be doing some Pr as proposals and we can discuss the detaisl within the Prs 😄

**Comment by HessTaha** [comment]
may I ask what's blocking this PR? #158

**Reply by TreyWW** [reply]
Something about poetry wasn't working, then i took a break and now I can't remember lol.

I'll have another look at it now, thanks

**Reply by TreyWW** [reply]
Just merged, thanks for the bump!
